### DISCUSSION TITLE
How to build swift ios toolchain?

### DISCUSSION THREAD

**Discussion by alexdoj** [theos/theos#615] [Questions]
(Thanks for reporting an issue! Please make sure you click the link above to view the issue guidelines, then fill out the blanks below.)

What are the steps to reproduce this issue?
-------------------------------------------
In docs described how to use an ios toolchain for Linux:
```
With Swift support (larger file size):

 sudo apt install zstd
 curl -LO https://github.com/CRKatri/llvm-project/releases/download/swift-5.3.2-RELEASE/swift-5.3.2-RELEASE-ubuntu18.04.tar.zst
 ....
```
https://github.com/theos/theos/wiki/Installation-Linux

It works for me, but I need to compile a swift toolchain for linux and I did next steps:

1. Downloaded and unarchived the sources from https://github.com/CRKatri/llvm-project/archive/refs/tags/swift-5.3.2-RELEASE.zip
2. 
```bash
cd llvm-project
mkdir build
cd build

cmake -G "Unix Makefiles" ../llvm \
  -DLLVM_TARGETS_TO_BUILD="X86;ARM;AArch64" \
  -DLLVM_ENABLE_PROJECTS="swift;clang" \
  -DLLVM_USE_LINKER=gold \
  -DCMAKE_BUILD_TYPE=Release
  
make
```

What happens?
-------------
There are no swift executables in the build dir, only clang/llvm stuff

What were you expecting to happen?
----------------------------------
expecting to contain swift executables (swift/swiftc)

Any logs, error output, etc?
----------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
…

What versions of software are you using?
----------------------------------------
**Operating System:** ubuntu 18.04

**Toolchain Version:** swift-5.3.2

**SDK Version:** -

**Comment by kirb** [comment]
Swift is separate from the LLVM project; you need to build it in addition to the base LLVM toolchain. The Swift project [readme](https://github.com/apple/swift#swift-toolchains) has build instructions.
